object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionChecker.java

```diff
@@ -35,10 +38,10 @@
   public static final String COUNT_MUST_CALL = "countMustCall";
 
   /**
-   * The number of expressions with must-call obligations that were checked. Incremented only if the
+   * Set containing expressions with must-call obligations that were checked. Increment only if the
    * {@link #COUNT_MUST_CALL} option was supplied.
    */
-  int numMustCall = 0;
+  Set<Tree> mustCallObligations = new HashSet<>();
```

Comment:
This `Set` of obligations should be an instance field in the `MustCallInvokedChecker`.  If we put it here, it will persist across the whole compilation, which is a potential memory leak of `Tree`s.  We should still keep the counter separately here.